docs/architecture/adr-005-support-multisig-chain-link.md

```diff
@@ -0,0 +1,147 @@
+`tx multisign` commands. To send a transaction to the node, the following process takes place: 
+1. the multisig account owners create a raw transaction file
+2. the threshold number of signers sign it to using the `tx sign` command
+3. all individual signatures are gathered and used as input from the `tx multisign` to generate the final signed transaction.
+
+The multisign file includes not only all public keys and threshold of the multisig account, but also the required number of signatures. Thanks to this, we can leverage all the information stored inside such file to create a proper chain link JSON starting from a multisigned transaction.
+
+The whole process in code is presented below:
+```go
+// getChainLinkJSONFromMultiSign generates the chain-link JSON from the multisign file and its raw transaction file
+func getChainLinkJSONFromMultiSign(
+	cmd *cobra.Command,
+	txFile string,
+	chainID string,
+	chain chainlinktypes.Chain,
+) (profilescliutils.ChainLinkJSON, error) {
+	clientCtx, err := client.GetClientTxContext(cmd)
+	if err != nil {
+		return profilescliutils.ChainLinkJSON{}, err
+	}
+
+	parsedTx, err := authclient.ReadTxFromFile(clientCtx, txFile)
+	if err != nil {
+		return profilescliutils.ChainLinkJSON{}, err
+	}
+
+	txCfg := clientCtx.TxConfig
+	txBuilder, err := txCfg.WrapTxBuilder(parsedTx)
+	if err != nil {
+		return profilescliutils.ChainLinkJSON{}, err
+	}
+
+	txFactory := tx.NewFactoryCLI(clientCtx, cmd.Flags())
+	if txFactory.SignMode() == signingtypes.SignMode_SIGN_MODE_UNSPECIFIED {
+		txFactory = txFactory.WithSignMode(signingtypes.SignMode_SIGN_MODE_LEGACY_AMINO_JSON)
+	}
+
+	sigs, err := txBuilder.GetTx().GetSignaturesV2()
+	if len(sigs) != 1 {
+		return profilescliutils.ChainLinkJSON{}, fmt.Errorf("invalid number of signatures")
+	}
+	multisigSig := sigs[0]
+
+	signingData := authSigning.SignerData{
+		ChainID:       chainID,
+		AccountNumber: txFactory.AccountNumber(),
+		Sequence:      txFactory.Sequence(),
+	}
+	// the bytes of plain text
+	value, err := txCfg.SignModeHandler().GetSignBytes(txFactory.SignMode(), signingData, txBuilder.GetTx())
+	if err != nil {
+		return profilescliutils.ChainLinkJSON{}, err
+	}
+
+	addr, _ := sdk.Bech32ifyAddressBytes(chain.Prefix, multisigSig.PubKey.Address().Bytes())
+	return profilescliutils.NewChainLinkJSON(
+		profilestypes.NewBech32Address(addr, chain.Prefix),
+		profilestypes.NewProof(multisigSig.PubKey, signingtypes.SignatureDataToProto(multisigSig.Data), hex.EncodeToString(value)),
+		profilestypes.NewChainConfig(chain.Name),
+	), nil
+}
+```
+
+## Consequences
+
+### Backwards Compatibility
+
+Since we change the `Proof#Signature` field from a hex-encode string into a `SignatureDescriptor_Data`, this feature is not backwards compatible. Migrating the old chain link signatures to `SignatureDescriptor_Data`
+is required.
```

Comment:
```suggestion
Since we change the `Proof#Signature` field from a hex-encoded string into a `SignatureDescriptor_Data`, this feature is not backwards compatible. For this reason, it will be needed to migrate the old chain link signatures to `SignatureDescriptor_Data`.
```